gtsam/hybrid/HybridBayesNet.cpp

```diff
@@ -52,15 +52,22 @@ HybridBayesNet HybridBayesNet::prune(size_t maxNrLeaves) const {
   // Multiply into one big conditional. NOTE: possibly quite expensive.
   DiscreteConditional joint;
   for (auto &&conditional : marginal) {
-    joint = joint * (*conditional);
+    // The last discrete conditional may be a TableDistribution
+    if (auto dtc = std::dynamic_pointer_cast<TableDistribution>(conditional)) {
+      DiscreteConditional dc(dtc->nrFrontals(), dtc->toDecisionTreeFactor());
+      joint = joint * dc;
+    } else {
+      joint = joint * (*conditional);
+    }
   }
 
-  // Prune the joint. NOTE: again, possibly quite expensive.
-  const DecisionTreeFactor pruned = joint.prune(maxNrLeaves);
+  // Prune the joint. NOTE: imperative and, again, possibly quite expensive.
+  DiscreteConditional pruned(joint);
+  pruned.prune(maxNrLeaves);
 
   // Create a the result starting with the pruned joint.
   HybridBayesNet result;
-  result.emplace_shared<DiscreteConditional>(pruned.size(), pruned);
+  result.push_back(std::make_shared<DiscreteConditional>(pruned));
```

Comment:
Probably better to first emplace then prune to avoid a copy
```
result.emplace_shared<DiscreteConditional>(joint);
result->back()->prune(maxNrLeaves);
```

Reply:
So I will have to do `DiscreteConditional pruned = *result.back()->asDiscrete()` because we need `pruned` for pruning the `HybridGaussianConditional`s.

I guess that didn't help with the extra copy :(

Reply:
No, I think results stores shared pointers, so that will work fine.